packages/ui/src/ui/components/JobStatisticTable/JobStatisticTable.tsx

```diff
@@ -0,0 +1,180 @@
+        return {minWidth, paddingLeft: (item?.level || 0) * LEVEL_OFFSET};
+    }, [item.level, minWidth]);
+
+    const toggleItemAndTreeState = useCallback(() => {
+        if (!itemState.empty) {
+            toggleItemState();
+        }
+    }, [itemState, toggleItemState]);
+
+    if (item.isLeafNode) {
+        return (
+            <span className={block('metric')} style={offsetStyle}>
+                <Icon awesome="chart-line" className={block('metric-icon')} />
+
+                <span>{renderValue(item)}</span>
+            </span>
+        );
+    } else {
+        return (
+            <span className={block('group')} style={offsetStyle} onClick={toggleItemAndTreeState}>
+                <Icon
+                    awesome={itemState.collapsed || itemState.empty ? 'angle-down' : 'angle-up'}
+                    className={block('group-icon-toggler')}
+                />
+                <Icon
+                    awesome={itemState.collapsed || itemState.empty ? 'folder' : 'folder-open'}
+                    className={block('group-icon')}
+                />
+                <span>{renderValue(item)}</span>
+            </span>
+        );
+    }
+}
+
+export type TableProps<Item extends Partial<JobTreeItem>> = {
+    columns?: {
+        mode: string;
+        sets?: {
+            default?: {
+                items: string[];
+            };
+        };
+        items: Record<
+            string,
+            {
+                sort: boolean;
+                align: string;
+            }
+        >;
+    };
+    theme: string;
+    size: string;
+    striped: boolean;
+    tree: boolean;
+    computeKey(item: Item): string;
+};
+
+export type JobStatisticTableTemplate<Item extends Partial<JobTreeItem>> = {
+    [name: string]: (
+        item: Item,
+        colName: string,
+        toggleItemState: Function,
+        itemState: ItemState,
+    ) => React.ReactChild | null;
+};
+
+export function GeneralJobStatisticTable<Item extends Partial<JobTreeItem>>({
+    items,
+    templates,
+    onFilterChange,
+    tableProps,
+}: {
+    items: Item[];
+    templates: JobStatisticTableTemplate<Item>;
+    onFilterChange(text: string): void;
+    tableProps: TableProps<Item>;
+}) {
+    const [treeState, setTreeState] = React.useState('expanded');
+
+    return (
+        <ErrorBoundary>
+            <div className={block()}>
+                <Toolbar onFilterChange={onFilterChange} onTreeStateChange={setTreeState} />
+
+                <div className={block('table-container')}>
+                    <ElementsTable
+                        {...tableProps}
+                        treeState={treeState}
+                        templates={templates}
+                        items={items}
+                        css={block()}
+                    />
+                </div>
+            </div>
+        </ErrorBoundary>
+    );
+}
+
+GeneralJobStatisticTable.Cell = JobStatisticTableStaticCell;
```

Comment:
When it is possible it is better to avoid such "static" `exports` at least, separate exports are good enough and more clear. 
I think we shouldn't export `JobStatisticTableStaticCell` and `ExpandedCell`, instead of it we should describe `templates` in the file